spring-xd-module/src/main/java/org/springframework/xd/module/spark/streaming/SparkStreamingDriverModule.java

```diff
@@ -273,8 +321,14 @@ public void stop() {
 		logger.info("stopping SparkDriver");
 		try {
 			// todo: when possible (spark 1.3.0), change to streamingContext.stop(false, true) without the cancel
-			streamingContext.ssc().sc().cancelAllJobs();
-			streamingContext.close();
+			if (javaStreamingContext != null) {
+				javaStreamingContext.ssc().sc().cancelAllJobs();
+				javaStreamingContext.close();
+			}
+			if (streamingContext != null) {
+				streamingContext.sc().cancelAllJobs();
+				streamingContext.stop(true, false);
```

Comment:
is this what we want here? The first boolean indicates stopping the actual SparkContext and the 2nd boolean is FALSE for "graceful"
So, two things jump out:
1) those boolean values are the reverse of what we mention in the TODO above this
2) this is different than the Java version (where we just call close), and that was done that way - along with the cancel call - as a temporary workaround until the proper stop (with false, true) would work